### ISSUE TITLE
Error occurs when generating variable fonts

### ISSUE THREAD

**Issue by RosaWagner** [googlefonts/fontmake#675]
I have an error when generating a variable font with Fontmake that I never had before, I don't know what it means:

`fontmake: Error: In 'sources/Arima.glyphs' -> 'master_ufo/Arima.designspace': Generating fonts from Designspace failed: ('Base master not found.', 'MarkBasePos', '[0]', 'Lookup', '[1]', 'list', '.Lookup', 'LookupList', '.LookupList', 'GPOS', '.table', 'table_G_P_O_S_')`

Static otf and ttf are correctly generated. The variable was exporting correctly before I add 2 indic scripts in the file.
Fork with source files is here : https://github.com/RosaWagner/Arima-Font

**Comment by punchcutter** [comment]
It's because of the `imatra` anchor on ssa-tamil. In the Bold weight the anchor is above y=500 so it gets put into the `abvm` lookup, but in the Light weight the anchor is below so it gets put into the `blwm` lookup. 

The relevant code is in ufo2ft.featureWriters.markFeatureWriter
```
    def _getVerticalThreshold(self):
        # anchors with unknown names whose Y coordinate is greater or equal to
        # the line that cuts the UPEM square in half will be treated as "above
        # base" marks, those that fall below the threshold as "below base".
        return getAttrWithFallback(self.context.font.info, "unitsPerEm") // 2
```

If the name is changed to any of these it will be put in the right place:
```
    abvmAnchorNames = {"top", "topleft", "topright", "candra", "bindu", "candrabindu"}
    blwmAnchorNames = {"bottom", "bottomleft", "bottomright", "nukta"}
```
But looking at that now it seems like it might be worth adding something like "iimatra". If you change that anchor to "topright" in all the Tamil glyphs and the iiMatra then it will work.

**Comment by RosaWagner** [comment]
Great! I will look into it and inform the authors. Thanks a lot :)

**Comment by anthrotype** [comment]
> looking at that now it seems like it might be worth adding something like "iimatra".

if Glyphs.app does that, then fontmake should as well (as that abvm/blwm feature generation was designed to match Glyphs.app).
Can you send a pull request to ufo2ft?

**Comment by anthrotype** [comment]
is it "imatra" or "iimatra"?

**Comment by punchcutter** [comment]
@anthrotype The mark is iiMatra, but the anchor here is named imatra. I'm not sure exactly what Glyphs is doing here, but with the imatra name it puts everything in `abvm` except tta-tamil which has the anchor near the baseline. It would make more sense to me to have some rules like if 'top' or 'bottom' are in the name then they go into `abvm` or `blwm`. If the anchors don't have that then why don't we check how many seem to be above and how many below and majority wins. In this case ssa-tamil and tta-tamil are the only glyphs where this anchor is below 500 so everything with this anchor should be put into the same `abvm` lookup.

**Comment by anthrotype** [comment]
> I'm not sure exactly what Glyphs is doing here

me neither. @schriftgestalt?

**Comment by schriftgestalt** [comment]
There is some heuristic regarding the position of the anchors. If they are in the top or bottom of the bounding box of the outline.

This is the first time someone complains about that behavior ;)

**Comment by anthrotype** [comment]
Yes, i think we implemented the same heuristic in ufo2ft. Would be interesting to know if the same issue occurs when exporting from Glyphs.app as well as from fontmake. @RosaWagner ?

**Comment by RosaWagner** [comment]
I can export the variable font with no fail with glyph app. Although I don’t know if the Tamil script works appropriately. I export a GX from glyphs and push it for you to look at it if you want.

**Comment by RosaWagner** [comment]
I just created a branch for testing this: https://github.com/RosaWagner/Arima-Font/tree/imatra-tests

**Comment by punchcutter** [comment]
@anthrotype i already said what Glyphs does in this case. It doesn’t separate this anchor on ssa-tamil like fontmake does. But fontmake is basing it on the top and bottom half of the UPM, not the bounding box of the glyph outline. That’s a huge difference.

**Comment by anthrotype** [comment]
Ah ok thanks. We'll use the bbox then

**Comment by eliheuer** [comment]
I've been running into a lot of `'Base master not found.', 'MarkBasePos',` errors recently. Will try digging into improving things around this area at some time this year hopefully.

**Comment by alexeiva** [comment]
I am having a similar issue in a private project with 2 mono masters and 3 proportional master. 
```
fontmake.errors.FontmakeError: In 'XXXX.glyphs' -> 
'master_ufo/XXXX.designspace': Generating fonts from 
Designspace failed: ((4, [3, 3, 4, 4, 4]), 'int', '.FeatureCount', 
'FeatureList','.FeatureList', 'GPOS', '.table', 'table_G_P_O_S_')
```
Static fonts build correctly, and Italic with 2 mono masters and 2 proportional master builds a VF fine. 

Features are set to a bare minimum:
![Screenshot 2021-03-05 at 10 36 45](https://user-images.githubusercontent.com/16634089/110082812-eb3ed380-7d9e-11eb-89a0-855df4fa9900.png)

What could be the potential cause of this?
